airsonic-main/src/main/java/org/airsonic/player/util/LegacyHsqlMigrationUtil.java

```diff
@@ -76,8 +76,8 @@ public static boolean isHsqlDbBackupNeeded(String dbPath, String jdbcUrl) {
                 return true;
             }
         } catch (SQLException e) {
-            LOG.warn("HSQLDB database driver version cannot be determined. Will make a copy of the DB prior to connecting. This may upgrade the database from version {}", currentVersion, e);
-            return true;
+            LOG.warn("HSQLDB database driver version cannot be determined.", e);
+            return false;
```

Comment:
This just stops it from making a backup.

This won't do anything to try and upgrade (or prevent it) any legacy dbs